src/csim/update_ops_control_single_target_single.cpp

```diff
@@ -251,3 +252,132 @@ void single_qubit_control_single_qubit_dense_matrix_gate_simd(
     }
 }
 #endif
+
+#ifdef _USE_MPI
+void single_qubit_control_single_qubit_dense_matrix_gate_mpi_OI(
+    UINT control_qubit_index, UINT control_value, CTYPE* t,
+    const CTYPE matrix[4], CTYPE* state, ITYPE dim, int flag,
+    UINT index_offset);
+void single_qubit_control_single_qubit_dense_matrix_gate_mpi_OO(
+    CTYPE* t, const CTYPE matrix[4], CTYPE* state, ITYPE dim, int flag);
+
+void single_qubit_control_single_qubit_dense_matrix_gate_mpi(
+    UINT control_qubit_index, UINT control_value, UINT target_qubit_index,
+    const CTYPE matrix[4], CTYPE* state, ITYPE dim, UINT inner_qc) {
+    MPIutil& m = MPIutil::get_inst();
+    const UINT rank = m.get_rank();
+    ITYPE dim_work = dim;
+    ITYPE num_work = 0;
+    CTYPE* t = m.get_workarea(&dim_work, &num_work);
+    assert(num_work > 0);
+    CTYPE* si = state;
+
+    if (target_qubit_index < inner_qc) {  // control, target: inner, inner
+        if (control_qubit_index < inner_qc) {
+            single_qubit_control_single_qubit_dense_matrix_gate(
+                control_qubit_index, control_value, target_qubit_index, matrix,
+                state, dim);
+        } else {  // control, target: outer, inner
+            const UINT control_rank_bit = 1 << (control_qubit_index - inner_qc);
+            if (((rank & control_rank_bit) && (control_value == 1)) ||
+                (!(rank & control_rank_bit) && (control_value == 0)))
+                single_qubit_dense_matrix_gate(
+                    target_qubit_index, matrix, state, dim);
+        }
+    } else {  // control, target: inner, outer
```

Comment:
small fix: This comment should be written in 6 lines below.

Reply:
Thanks, I corrected it.